Build failure: sv-lang

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix shell nixpkgs#sv-lang`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/308928727

### Relevant log output

```console
error: Cannot build '/nix/store/kasxipqiybwr7jfgfr5jcmalk4kjjlj2-sv-lang-7.0.drv'.
       Reason: builder failed with exit code 8.
       Output paths:
         /nix/store/ji5204mbwmhpxq9l06kbyx2m0f3r616k-sv-lang-7.0
       Last 25 log lines:
       >     Start 1: unittests
       >     Start 2: regression_delayed_reg
       >     Start 3: regression_wire_module
       >     Start 4: regression_all_file
       >     Start 5: netlist_unittests
       >     Start 6: tidy_unittests
       > 1/6 Test #3: regression_wire_module ...........   Passed    0.00 sec
       > 2/6 Test #2: regression_delayed_reg ...........   Passed    0.00 sec
       > 3/6 Test #5: netlist_unittests ................   Passed    0.00 sec
       > 4/6 Test #6: tidy_unittests ...................   Passed    0.01 sec
       > 5/6 Test #4: regression_all_file ..............   Passed    0.01 sec
       > 6/6 Test #1: unittests ........................***Timeout  60.06 sec
       > Randomness seeded to: 1399792561
       >
       >
       > 83% tests passed, 1 tests failed out of 6
       >
       > Total Test time (real) =  60.06 sec
       >
       > The following tests FAILED:
       >        1 - unittests (Timeout)
       > Errors while running CTest
       > FAILED: [code=8] CMakeFiles/test.util
       > cd /build/source/build && /nix/store/2lrbixyw95bjg0x1aav648r0h0zsj2jl-cmake-4.1.1/bin/ctest
       > ninja: build stopped: subcommand failed.
       For full logs, run:
         nix log /nix/store/kasxipqiybwr7jfgfr5jcmalk4kjjlj2-sv-lang-7.0.drv
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.7, NixOS, 25.11 (Xantusia), 25.11.20251012.cf3f5c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/la80d03vcf555bkcjp84cij5c4i63m5n-source`

### Notify maintainers

@SharzyL 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/sv/sv-lang/package.nix
+++ b/pkgs/by-name/sv/sv-lang/package.nix
@@ -13,18 +13,18 @@
 
 stdenv.mkDerivation rec {
   pname = "sv-lang";
-  version = "7.0";
+  version = "9.1";
 
   src = fetchFromGitHub {
     owner = "MikePopoloski";
     repo = "slang";
     rev = "v${version}";
-    sha256 = "sha256-msSc6jw2xbEZfOwtqwFEDIKcwf5SDKp+j15lVbNO98g=";
+    sha256 = "sha256-IfRh6F6vA+nFa+diPKD2aMv9kRbvVIY80IqX0d+d5JA=";
   };
 
   postPatch = ''
     substituteInPlace external/CMakeLists.txt \
-      --replace-fail 'set(mimalloc_min_version "2.1")' 'set(mimalloc_min_version "${lib.versions.majorMinor mimalloc.version}")'
+      --replace-fail 'set(mimalloc_min_version "2.2")' 'set(mimalloc_min_version "${lib.versions.majorMinor mimalloc.version}")'
   '';
 
   cmakeFlags = [

--- a/pkgs/by-name/sv/sv-lang/package.nix
+++ b/pkgs/by-name/sv/sv-lang/package.nix
@@ -11,20 +11,21 @@
   python3,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "sv-lang";
   version = "9.1";
 
   src = fetchFromGitHub {
     owner = "MikePopoloski";
     repo = "slang";
-    rev = "v${version}";
-    sha256 = "sha256-IfRh6F6vA+nFa+diPKD2aMv9kRbvVIY80IqX0d+d5JA=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-IfRh6F6vA+nFa+diPKD2aMv9kRbvVIY80IqX0d+d5JA=";
   };
 
   postPatch = ''
-    substituteInPlace external/CMakeLists.txt \
-      --replace-fail 'set(mimalloc_min_version "2.2")' 'set(mimalloc_min_version "${lib.versions.majorMinor mimalloc.version}")'
+    substituteInPlace external/CMakeLists.txt --replace-fail \
+      'set(mimalloc_min_version "2.2")' \
+      'set(mimalloc_min_version "${lib.versions.majorMinor mimalloc.version}")'
   '';
 
   cmakeFlags = [
@@ -32,7 +33,7 @@ stdenv.mkDerivation rec {
     "-DCMAKE_INSTALL_INCLUDEDIR=include"
     "-DCMAKE_INSTALL_LIBDIR=lib"
 
-    "-DSLANG_INCLUDE_TESTS=${if doCheck then "ON" else "OFF"}"
+    "-DSLANG_INCLUDE_TESTS=${if finalAttrs.finalPackage.doCheck then "ON" else "OFF"}"
   ];
 
   nativeBuildInputs = [
@@ -40,6 +41,8 @@ stdenv.mkDerivation rec {
     python3
     ninja
   ];
+
+  strictDeps = true;
 
   buildInputs = [
     boost
@@ -62,4 +65,4 @@ stdenv.mkDerivation rec {
     platforms = platforms.all;
     broken = stdenv.hostPlatform.isDarwin;
   };
-}
+})

